src/session.ts

```diff
@@ -238,4 +224,19 @@ export namespace DebugSession {
      */
     client: IClientSession | Session.ISession;
   }
+
+  /**
+   * Request whether debugging is enabled for the given client.
+   * @param client The client session.
+   */
+  export async function requestDebuggingEnabled(
+    client: IClientSession | Session.ISession
+  ) {
```

Comment:
This function should have a return type: `Promise<boolean>` and a return statement that guarantees a boolean is returned: `return !!(info?.debugger ?? false);`

Reply:
Sounds good :+1: 

It's true that kernels can still return an empty string for example for the `debugger` key.

Reply:
Sure, but we will have to assume that's `false`.

This might belong in the `IDebugger` interface so others can consume it, as a static method in that namespace ... `IDebugger#isAvailable(client)` maybe? It seems like the kind of thing consumers of the `IDebugger` might care about?

Reply:
> It's true that kernels can still return an empty string for example for the `debugger` key.

I realize that sentence was maybe a bit confusing. What I meant is that we indeed need to make sure the return type is a `boolean` because for now the `debugger` key is not part of the protocol yet, so we could get other kinds of values. So +1 to enforce the `boolean` type.

> This might belong in the `IDebugger` interface so others can consume it, as a static method in that namespace ... `IDebugger#isAvailable(client)` maybe? It seems like the kind of thing consumers of the `IDebugger` might care about?

That could indeed be part of the public API.

Reply:
It looks like TS would not allow defining a static method (exported function) in that namespace if we try to follow that pattern: 

https://www.typescriptlang.org/play/index.html?ssl=15&ssc=6&pln=13&pc=1#code/CYUwxgNghgTiAEkoGdnwCoHsDWIB2APOgHzwDeAsAFDy2KZ7IAuMArmE5jABR5QC2IAFzxmMAJZ4A5gEoA3NQC+1FVRAAPAA5cm8SUxAwAZlDAIAkgBEQAI1ZSph8tTrw4UYAwgBPPcgDKTLAGwCI2mJgQIFB4CjR0zMHcMiIACjCY-OLIIAQAbpjiwMRxrsyYmslpGVk5+YXFccpU1BraMLp8gsiaphbWdg5OlPG0bTrwRqx4HOIMfgCCeVDi0DZRyc6jrm4gTKwwePAsrCCldM3NrVoTYAzM8Fa29o4w8AC88HggAO4YOPgCE9Bq9iNwAEQAAQAVqxNN4DDA1gB6UAgwzg+RAA

Reply:
Done as a method in `DebugService` and exposed via `IDebugger`.